pkg/cmd/root/root.go

```diff
@@ -33,18 +38,33 @@ func ConfigureGlobalLogger(logLevelFlag string) {
 			Level: log.PanicLevel,
 		}
 	}
-
+	log.Debug().
+		Str("version", versionInfo.Version).
+		Str("commit", versionInfo.Commit).
+		Str("date", versionInfo.Commit).
+		Msgf("starting up, args: %+v", os.Args[1:])
 	log.Trace().Msgf("Log level set to %s", logLevelFlag)
 }
 
+func isCompletion() bool {
+	return len(os.Args) >= 1 && os.Args[1] == "__complete"
+}
+
 func NewRootCmd(cfg *config.Config) *cobra.Command {
 	var logLevel string
+	var nonInteractive bool
+
 	rootCmd := &cobra.Command{
 		Use:   "corectl",
 		Short: "CLI interface for the CECG core platform.",
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
-			ConfigureGlobalLogger(logLevel)
-			cmd.SilenceErrors = true
+			if !isCompletion() {
+				ConfigureGlobalLogger(logLevel)
+				if cmd.Name() != "update" {
```

Comment:
Let's extract it to a global constant in cmd/update.go and reuse it here and there

Reply:
it's only needed in root, we haven't actually called the cmd at this point

Reply:
I don't think I understand your point.
You don't have to call anything to use a constant